src/IMDb.Importer/ImdbDatasetImporter.cs

```diff
@@ -110,99 +110,53 @@ private async Task<string> DownloadDataset(string url, string outputFileName)
             return fileName;
         }
 
-        // These could be refactored to have the shared logic in one spot, but its not that much code, so I don't mind for now
-
         #region Individual Dataset Import Functions
 
         private async Task DownloadAndImportNameBasics()
         {
-            var file = await DownloadDataset("https://datasets.imdbws.com/name.basics.tsv.gz", nameof(NameBasics));
-            await using var fileStream = File.OpenRead(file);
-
-            var batches = _datasetParser.Parse<NameBasics>(fileStream);
-
-            await foreach (var batch in batches)
-            {
-                _nameBasicsRepository.BulkSync(batch);
-            }
+            await DownloadAndImportType<NameBasics>(_nameBasicsRepository);
```

Comment:
Instead of having 1 function call 1 function, these `DownloadAndImportType<T>()` calls should be just in the task list `DownloadAndImportAll()`